plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/conf/__init__.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+from omegaconf import MISSING
+
+
+@dataclass
+class RayInitConf(Dict[str, Any]):
+    """ parameters for ray.init()"""
+
+    pass
+
+
+@dataclass
+class RayRemoteConf(Dict[str, Any]):
+    """ parameters for @ray.remote() """
+
+    num_cpus: Optional[int] = 1
+    num_gpus: Optional[int] = 0
+
+
+@dataclass
+class RayConf:
+    ray_init_cfg: RayInitConf = field(default_factory=RayInitConf)
+    ray_remote_cfg: RayRemoteConf = field(default_factory=RayRemoteConf)
+
+
+@dataclass
+class RayLocalLauncherConf(ObjectConf):
+    params: RayConf = RayConf()
+
+
+# Ray AWS config
+
+
+class RayAutoScalingMode(Enum):
+    default = "default"
+    aggressive = "aggressive"
+
+
+@dataclass
+class RayDockerConf(Dict[str, Any]):
+    """
+    This executes all commands on all nodes in the docker container,
+    and opens all the necessary ports to support the Ray cluster.
+    Empty string means disabled.
+
+    """
+
+    # e.g., tensorflow/tensorflow:1.5.0-py3
+    image: str = ""
+    # e.g. ray_docker
+    container_name: str = ""
+    # If true, pulls latest version of image. Otherwise, `docker run` will only pull the image
+    # if no cached version is present.
+    pull_before_run: bool = True
+    # Extra options to pass into "docker run"
+    run_options: List[str] = field(default_factory=list)
+
+
+@dataclass
+class RayProviderConf(Dict[str, Any]):
+    type: str = "aws"
+    region: str = "us-west-2"
+    availability_zone: str = "us-west-2a,us-west-2b"
+    cache_stopped_nodes: bool = False
+
+
+@dataclass
+class RayAuthConf(Dict[str, Any]):
+    ssh_user: str = "ubuntu"
+
+
+@dataclass
+class RayNodeConf(Dict[str, Any]):
+    """
+    Additional options in boto docs.
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html
+    """
+
+    InstanceType: str = "m5.large"
+    ImageId: str = "ami-008d8ed4bd7dc2485"
```

Comment:
You are extending Dict everywhere. make a full cleanup pass on it.
I want a reason for every Dict extention left.

Reply:
there's many other fields a user can pass in other than the ones listed here (see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html)

I think we need to extend Dict to support  this sort of scenario?